### ISSUE TITLE
Invalid CS8629 after nullable reference type comparison

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn#37039]
**Version Used**: 3.3.0-beta1-19327-04

**Steps to Reproduce**:

```c#
int FindLargest(IEnumerable<int?> ints)
{
    int largest = int.MinValue;

    foreach (int? i in ints)
    {
        if (i > largest)
        {
            largest = i.Value; // CS8629 Nullable value type may be null
        }
    }

    return largest;
}
```

This requires the loop to trigger. This simple form doesn't have a diagnostic:

```c#
int M(int? i) => i > 0 ? i.Value : 0;
```

**Expected Behavior**:

Compiler learns that `i` is not null when integer comparison succeed and does not produce a diagnostic.

**Actual Behavior**:

CS8629 diagnostic.

**Comment by jcouv** [comment]
Seems like we've fixed this ([sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+BiAdgVwBtCJhC4ACOXU8gWACgABABgqYEYAWAbkcaYAmCgGFGAb0YBLXDAoAxGSgAyEBAHM4sABScAzAB4ZMAPwA+CsagBKCYwoPLsiiQ1a5AXicwAdAFkZADUIQnw4PgZ7RwAzAHsEOAgAYwALCm1jE0snbxsoh0kGR2LLaPSpCgtXTVhbIpKC/IbHavcKLykfYNDwigB6PtEAZQAOADZBAE4KADkiEjJKADcQsIoYAE8AB0pIDYpgSgJiJuKAXyaLyPqHJgB2FzUamAirs6A=))
